tracing/src/macros.rs

```diff
@@ -2017,6 +2063,21 @@ macro_rules! fieldset {
         $crate::fieldset!(@ { $($out),*, $crate::__tracing_stringify!($($k).+) } $($rest)*)
     };
 
+    (@ { $(,)* $($out:expr),* } $k:literal = ?$val:expr, $($rest:tt)*) => {
+        $crate::fieldset!(@ { $($out),*, $crate::__tracing_stringify!($k) } $($rest)*)
+    };
+    (@ { $(,)* $($out:expr),* } $k:literal = %$val:expr, $($rest:tt)*) => {
+        $crate::fieldset!(@ { $($out),*, $crate::__tracing_stringify!($k) } $($rest)*)
+    };
+    (@ { $(,)* $($out:expr),* } $k:literal = $val:expr, $($rest:tt)*) => {
+        $crate::fieldset!(@ { $($out),*, $crate::__tracing_stringify!($k) } $($rest)*)
+    };
```

Comment:
I don't think this is correct. The `stringify!` compiler built-in macro is used here to convert tokens that are identifiers (or dotted sequences of identifiers) to string literals. Using `stringify!` on a token that's already a string literal will create a string literal containing the _string representation_ of that token: it will contain quotation marks and escape sequences for any escaped characters. For example, see [this playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=4760fe1b9865355171deb81aac740bbc).

I think instead of stringifying fields whose names are already string literals, we just want to use the provided string literal directly:
```suggestion
    (@ { $(,)* $($out:expr),* } $k:literal = ?$val:expr, $($rest:tt)*) => {
        $crate::fieldset!(@ { $($out),*, $k } $($rest)*)
    };
    (@ { $(,)* $($out:expr),* } $k:literal = %$val:expr, $($rest:tt)*) => {
        $crate::fieldset!(@ { $($out),*, $k } $($rest)*)
    };
    (@ { $(,)* $($out:expr),* } $k:literal = $val:expr, $($rest:tt)*) => {
        $crate::fieldset!(@ { $($out),*, $k) } $($rest)*)
    };
```

Reply:
This suggestion doesn't compile

Reply:
There's a typo in the suggestion, but yes, done.